libs/filamat/src/MaterialBuilder.cpp

```diff
@@ -376,13 +379,20 @@ bool MaterialBuilder::runStaticCodeAnalysis() noexcept {
 #ifndef FILAMAT_LITE
     GLSLTools glslTools;
 
+    // Some fields in MaterialInputs only exist if the property is set (e.g: normal, subsurface
+    // for cloth shading model). Give our shader all properties. This will enable us to parse and
+    // static code analyse the AST.
+    MaterialBuilder::PropertyList allProperties;
+    std::fill_n(allProperties, MATERIAL_PROPERTIES_COUNT, true);
+    ShaderModel model;
+    std::string shaderCodeAllProperties = peek(ShaderType::FRAGMENT, model, allProperties);
```

Comment:
Is peeking here equivalent to what we were doing before? A peek generates the shader and should be avoided whenever possible.

Reply:
Yes. Before, [`findProperties` did the `peek`](https://github.com/google/filament/pull/1254/files#diff-32d2e5085c15950b21f6319ae615b9e8L155). Now, we do it inside `MaterialBuilder`, and pass the result string to `findProperties`. This change doesn't affect performance, but makes the logic simpler when material domains are introduced.

 I agree peeking should be avoided and might be possible to remove in the future.

Reply:
Sounds good thanks